### PR TITLE
KAFKA-6826 avoid range scans when forwarding values during aggregation

### PR DESCRIPTION
**Pull request by xvrl**
@guozhangwang

### CODE REVIEW

File: streams/src/main/java/org/apache/kafka/streams/state/internals/CachingWindowStore.java [hunk 1]
```diff
@@ -219,13 +219,11 @@ public synchronized void put(final Bytes key, final byte[] value, final long tim
     }
     
     private V fetchPrevious(final Bytes key, final long timestamp) {
-        try (final WindowStoreIterator<byte[]> iter = underlying.fetch(key, timestamp, timestamp)) {
-            if (!iter.hasNext()) {
-                return null;
-            } else {
-                return serdes.valueFrom(iter.next().value);
-            }
+        byte[] value = underlying.fetch(key, timestamp);
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/state/internals/CachingWindowStore.java]: nit: add `final`

**Review by mjsax — COMMENTED** [review]
Thanks for the PR!

**Comment by mjsax** [comment]
\cc @bbejeck @vvcephei

**Review by bbejeck — COMMENTED** [review]
Thanks @xvrl , LGTM

**Comment by bbejeck** [comment]
retest this please

**Comment by vvcephei** [comment]
LGTM to me as well. Nice catch, @xvrl .

**Review by guozhangwang — APPROVED** [review]

**Pull request merged**